### ISSUE TITLE
Add Omit<T,K> and Diff<T,U> to lib.d.ts

### ISSUE THREAD

**Issue by Pajn** [Microsoft/TypeScript#19569]
As spread and rest types is taking time I usually have to copy these types in projects. They solve the primary use case for spread/rest types and works today. I think they should fit right in next to Pick and Partial.
And even when (if?) spread and rest types is merged these would still arguably have some value.

```ts
export type Diff<T extends string, U extends string> = ({[P in T]: P} &
  {[P in U]: never} & {[x: string]: never})[T]
export type Omit<T, K extends keyof T> = Pick<T, Diff<keyof T, K>>
```

**Comment by kitsonk** [comment]
Awesome type operators... even if it doesn't get included thanks for the ideas.

**Comment by Pajn** [comment]
Oh, maybe I should have made it clear. These are not mine, I found them buried deep in a conversation once but I managed to find them again here: https://github.com/Microsoft/TypeScript/issues/12215#issuecomment-307871458

It also have this great type which I had totally forgotten as it's hard to find that specific comment via Google.
```ts
type Overwrite<T, U> = { [P in Diff<keyof T, keyof U>]: T[P] } & U;
```

**Comment by NN---** [comment]
Please also ['Required'](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/20774) type from discussion of #15012

**Comment by mhegazy** [comment]
As i noted in the `Required` one, i think we need a type operator here instead.

**Comment by NN---** [comment]
@mhegazy Without any doubt it will be a great solution, but if it won't go in the foreseen TS version, then Required can be a nice workaround meanwhile.

**Comment by ahejlsberg** [comment]
See #21316.

**Comment by Pajn** [comment]
@ahejlsberg Conditional types would be awesome and solve many problems but this issue is just about integrating types already writeable (and already written in many `@types/` packages) in a default `.lib.d.ts`.

**Comment by ahejlsberg** [comment]
@Pajn Understood. The link was merely to point out there's now a better way to implement these types.

**Comment by Pajn** [comment]
Ah, yes that's possible. I just wanted to be clear that I don't think this issue should be closed by #21316. I don't want to sound negative though, I'm super excited by conditional types in itself!
